### PR TITLE
HHH-18019 fix package present detection on ecj

### PR DESCRIPTION
**Pull request by gavinking**
https://hibernate.atlassian.net/browse/HHH-18019

### CODE REVIEW

**Comment by yrodiere** [comment]
Backported to branch 6.5 as https://github.com/hibernate/hibernate-orm/pull/8435

File: tooling/metamodel-generator/src/main/java/org/hibernate/processor/HibernateProcessor.java [hunk 1]
```diff
@@ -251,6 +252,12 @@ private boolean handleSettings(ProcessingEnvironment environment) {
 		return parseBoolean( options.get( FULLY_ANNOTATION_CONFIGURED_OPTION ) );
 	}
 
+	private static boolean packagePresent(@Nullable PackageElement pack) {
+		return pack != null
+			//HHH-18019 ejc always returns a non-null PackageElement
+			&& !pack.getEnclosedElements().isEmpty();
```
Comment (famod) [inline tooling/metamodel-generator/src/main/java/org/hibernate/processor/HibernateProcessor.java:258]: Ha, nice - was just about to propose the same workaround (coming from https://github.com/quarkusio/quarkus/issues/40809).

@gavinking @yrodiere Are you going to report a bug for ECJ by any chance?

Comment (yrodiere) [reply tooling/metamodel-generator/src/main/java/org/hibernate/processor/HibernateProcessor.java:258]: @famod I wasn't intending to as I suspect there's a reason for this (we'd need to check past bug reports) and I'm traveling this week anyway, but if you can that'd be great!

Comment (gavinking) [reply tooling/metamodel-generator/src/main/java/org/hibernate/processor/HibernateProcessor.java:258]: @famod The Javadoc for that method isn't well-written, but it does _imply_ that the method is supposed to search for "non-empty packages".

**Pull request merged**